.claude/hooks/post-edit-compile.sh

```
@@ -0,0 +1,44 @@
+#
+# If the edited file is a .java source file the hook runs:
+#   ./gradlew compileJava --quiet
+# A non-zero exit feeds the compiler errors back to Claude in the same turn
+# (exit code 2), so it can fix compilation failures immediately.
+
+set -uo pipefail
+
+input=$(cat)
+
+# Extract tool_input.file_path from the JSON payload.
+file_path=$(python3 - <<<"$input" 2>/dev/null <<'PYEOF'
```

Shellcheck (0.11.0)

[error] Multiple redirections compete for stdin. Use cat, tee, or pass filenames instead.

(SC2261)